paddle/fluid/operators/dropout_op.cu

```diff
@@ -24,21 +24,11 @@ namespace paddle {
 namespace operators {
 
 template <typename T>
-__global__ void RandomGenerator(const size_t n, const int seed,
-                                const float dropout_prob, const T* src,
-                                T* mask_data, T* dst) {
-  thrust::minstd_rand rng;
-  rng.seed(seed);
-  thrust::uniform_real_distribution<float> dist(0, 1);
-
+__global__ void RandomGenerator(const size_t n, const T* src,
+                                const T* cpu_mask_data, T* mask_data, T* dst) {
   int idx = blockDim.x * blockIdx.x + threadIdx.x;
   for (; idx < n; idx += blockDim.x * gridDim.x) {
-    rng.discard(idx);
```

Comment:
Why does the old implementation do this?

Reply:
If we don't do the discard, then every thread will get the same random value. Now our random array are generated in CPU vector, so every thread will get the correct value without discarding any preceding numbers.